libavpipe/src/avpipe_xc.c

```diff
@@ -4605,6 +4532,12 @@ check_params(
         return eav_param;
     }
 
+    if (params->copy_mpegts) {
+        if (strcmp(params->format, "fmp4-segment")) {
```

Comment:
This is good. I'm glad to see this :)

Reply:
Noting here more validation is needed _after_ we probe the input stream:  we would find the type of stream (MPEGTS, SRT, etc) as well as frame rate.

We also need a 'stats' callback after that so the downstream application can also validate itself once the input stream is known.